include/fmt/core.h

```diff
@@ -392,11 +399,8 @@ template <typename Char> class basic_string_view {
     the size with ``std::char_traits<Char>::length``.
     \endrst
    */
-#if __cplusplus >= 201703L  // C++17's char_traits::length() is constexpr.
-  FMT_CONSTEXPR
-#endif
-  FMT_INLINE basic_string_view(const Char* s) : data_(s) {
-    if (std::is_same<Char, char>::value && !detail::is_constant_evaluated())
+  FMT_CONSTEXPR_CHAR_TRAITS_LENGTH FMT_INLINE basic_string_view(const Char* s) : data_(s) {
+    if FMT_CONSTEXPR_CHAR_TRAITS_LENGTH (std::is_same<Char, char>::value && !detail::is_constant_evaluated())
```

Comment:
This line becomes wrong with `FMT_CONSTEXPR_CHAR_TRAITS_LENGTH` replaced as `FMT_CONSTEXPR`, i.e. `constexpr`. From [cppreference](https://en.cppreference.com/w/cpp/types/is_constant_evaluated) about `std::is_constant_evaluated()` (which is used inside `detail::is_constant_evaluated()` when supported):

> When directly used as the condition of static_assert declaration or **_constexpr if_** statement, std::is_constant_evaluated() always returns true.

So it shouldn't be `if constexpr`, just `if`.

Reply:
The correct way to fix this warning I can imagine (probably it's also very unclear): combine `__cplusplus` check together with `__cpp_lib_is_constant_evaluated` check to use `if constexpr` only if `detail::is_constant_evaluated()` unconditionally false.

Reply:
Just replace `FMT_CONSTEXPR_CHAR_TRAITS_LENGTH` with `detail::const_check` and revert other changes.